r_d26/GameWorld.cpp

```diff
@@ -56,34 +59,94 @@ bool checkCollision(const sf::Sprite* sprite1, const sf::Sprite* sprite2)
     return false;
 }
 
+//void GameWorld::update(float DeltaTime)
+//{
+//    //Update all game objects
+//    for (GameObject* object : m_gameObjects)
+//    {
+//        object->update(DeltaTime);
+//    }
+//
+//    //Check Collisions
+//    bool isAnyOfCollided = false;
+//
+//    for (GameObject* box : m_collidableObjects)
+//    {
+//        if (checkCollision(player.getSprite(), box->getSprite()))
+//        {
+//            isAnyOfCollided = true;
+//            box->getSprite()->setColor(sf::Color::Red);
+//        }
+//        else
+//        {
+//            // This color is modulated (multiplied) with the sprite's texture.
+//            box->getSprite()->setColor(sf::Color::White);
+//        }
+//    }
+//    m_collisionCounter.update(isAnyOfCollided);
+//}
 
 void GameWorld::update(float DeltaTime)
 {
-    //Update all game objects
-    for (GameObject* object : m_gameObjects)
-    {
-        object->update(DeltaTime);
+    size_t numObjects = m_gameObjects.size();
+    size_t numCollidables = m_collidableObjects.size();
+    size_t numThreads = 2; // Number of threads to use or std::thread::hardware_concurrency();
+    std::vector<std::thread> threads;
+    std::mutex collisionMutex;
+    bool isAnyOfCollided = false;
+
+    // Divide work for updating game objects
+    for (size_t i = 0; i < numThreads; ++i) {
+        size_t start = i * (numObjects / numThreads);
+        size_t end = (i == numThreads - 1) ? numObjects : (i + 1) * (numObjects / numThreads);
+        threads.emplace_back(&GameWorld::updateGameObjects, this, DeltaTime, start, end);
+    }
+    for (auto& t : threads) {
+        t.join();
     }
```

Comment:
Creating new threads EVERY update, that's why it's not that impressive. The better way would be to create a bunch of threads in init and let them execute the same update all the time, but placing some kind of mutex at the end so it would wait the start of next tick. So there'll be no need to recreate threads all the time

Reply:
Conditional variables may be used to signal updates to continue their job once the next tick has come